### ISSUE TITLE
Feature Request: URL collector

### ISSUE THREAD

**Issue by axelsimon** [vector-im/riot-web#8746]
### Description

Conversations (rooms) in Riot currently have a way to list members of the conversations, notifications and files uploaded to the conversation. In line with this last feature, I think it would be very convenient to have a URL collector (aka URL grabber, URL snatcher, and other keywords others might use when searching the issues :wink:).

The idea is very simple: any link that is pasted to the conversation is collected by the URL collector and a new section next to "files" is shown: "links", that displays all the collected URLs.
Even in this simplistic version, it would already be very useful, as in ongoing and regular conversations, lost of links get shared and it is sometimes hard to go back to them.

A more advanced version would:
- display they date of the collected URLs
- organise them by week, month, year
- recognise protocols beyond HTTP and HTTPS like magnet links, IPFS or ZeroNet

For those who remember, this is a feature that Trillian, a multi-protocol IM client of the early noughties, had :slightly_smiling_face:

**Comment by axelsimon** [comment]
Not really a dev, but is there anyway I could help on this?

**Comment by t3chguy** [comment]
@axelsimon you could sponsor it: https://www.bountysource.com/teams/vector-im or similar

This could be implemented entirely in riot where it'd be far from efficient, or it could be written into the Matrix spec, which would require a lot more planning and discussions but would be a much better solution overall

**Comment by olymk2** [comment]
This would be awesome to see, something I keep thinking would be very handy to save scrolling for links so much :p

**Comment by tatusamma** [comment]
Absolutely, irssi and weechat had great plugins to enable this as well. This would be an awesome feature.
